cfpb/cfgov-refresh
# Mega menu: Add aria attributes to footer text

Mega menu footer links are often brief and rely on the surrounding text to be understood, so this automatically adds aria attributes to provide that context to users accessing the links via screen readers.

## Additions

- `NavFooter` block 
- MenuItem method that adds an `aria-label` attribute containing the full text of the paragraph to footer links and `aria-hidden` attributes to the surrounding text

## Changes

- Remove all editor options except paragraphs and links from mega menu nav footer content field

## Notes
- Automatically adding aria attributes depends on footer items being written so that the full text is usable as an aria-label on the link, which would necessitate a slight change to the text of the `Ask CFPB` link under `Consumer Tools`. 


## Testing

1. Check out branch
2. Run `python cfgov/manage.py migrate` 
3. Run`python cfgov/manage.py runscript migrate_mega_menu` if necessary to generate the mega menu
4. Turn on `WAGTAIL_MENU` flag to activate the Wagtail version of the mega menu 
5. Save a menu item (with or without edits to footer)
6. Open the mega menu and verify that aria attributes have been added to the footer items

## Screenshots
<img width="828" alt="screen shot 2018-06-19 at 9 23 44 am" src="https://user-images.githubusercontent.com/778171/41610664-a80bb74e-73a2-11e8-8339-81528ceac470.png">



## Checklist


## Testing checklist

### Browsers


### Accessibility


### Other
